code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/util/ServiceDiscoveryUtil.java

```diff
@@ -436,19 +438,30 @@ public static void injectBalancerLabelsAndHealthcheck(Map<Object, Object> launch
     private static boolean doesLBHaveDrainSupport(Map<Object, Object> launchConfig) {
         if(launchConfig.get(InstanceConstants.FIELD_IMAGE_UUID) != null) {
```

Comment:
Nit: it would be cleaner to return to have  if(launchConfig.get(InstanceConstants.FIELD_IMAGE_UUID) == null)  {
return false
}

and then have the rest of the code outside of the if/else block